File: Core/Entities/Product.cs
namespace Core.Entities;

public class Product : BaseEntity
{
    public string Name { get; set; }
    public string Description { get; set; }
    public double Price { get; set; }
    public string PictureUrl { get; set; }
    public int ProductTypeId { get; set; } // Foreign key for ProductType
    public int ProductBrandId { get; set; } // Foreign key for ProductBrand
    // Navigation properties
    public ProductType ProductType { get; set; }
    public ProductBrand ProductBrand { get; set; }
}

File: Client/src/app/shared/models/Product.ts
export interface Product {
  id: number;
  name: string;
  description: string;
  price: number;
  pictureUrl: string;
  productType: {
    id: number;
    name: string;
  };
  productBrand: {
    id: number;
    name: string;
  };
}
